homeassistant/components/google_photos/media_source.py

```diff
@@ -0,0 +1,291 @@
+"""Media source for Google Photos."""
+
+from dataclasses import dataclass
+import logging
+from typing import Any, cast
+
+from homeassistant.components.media_player import MediaClass, MediaType
+from homeassistant.components.media_source import (
+    BrowseError,
+    BrowseMediaSource,
+    MediaSource,
+    MediaSourceItem,
+    PlayMedia,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from . import GooglePhotosConfigEntry
+from .const import DOMAIN
+from .exceptions import GooglePhotosApiError
+
+_LOGGER = logging.getLogger(__name__)
+
+# Media Sources do not support paging, so we only show a subset of recent
+# photos when displaying the users library. We fetch a minimum of 50 photos
+# unless we run out, but in pages of 100 at a time given sometimes responses
+# may only contain a handful of items Fetches at least 50 photos.
```

Comment:
What i'm trying to hack around is if we set PAGE_SIZE = 50 you can get back like 10 photos (presumably due to server side filtering) and as a result spend a bunch of time on server side round trips. Given that context, have a suggestion for naming?

Reply:
Yes I understand the potential server side filtering that might return less photos than requested in the page.
But there is a punctuation missing in the comment, I think a period before "Fetches at least 50 photos", and it's confusing having MAX_ in the constant (and now max_photos in SpecialAlbumDetails) when it's actually MIN_, but only if you have that many photos in your library, otherwise it's a MAX_. Unfortunately I don't have a good suggestion.